akka-docs/rst/java/persistence-query.rst

```diff
@@ -0,0 +1,236 @@
+.. _persistence-query-java:
+
+#################
+Persistence Query
+#################
+
+Akka persistence query complements :ref:`persistence-java` by providing a universal asynchronous stream based
+query interface that various journal plugins can implement in order to expose their query capabilities.
+
+The most typical use case of persistence query is implementing the so-called query side (also known as "read side")
+in the popular CQRS architecture pattern - in which the writing side of the application (e.g. implemented using akka
+persistence) is completely separated from the "query side". Akka Persistence Query itself is *not* directly the query
+side of an application, however it can help to migrate data from the write side to the query side database. In very
+simple scenarios Persistence Query may be powerful enough to fulful the query needs of your app, however we highly
+recommend (in the spirit of CQRS) of splitting up the write/read sides into separate datastores as the need arrises.
+
+While queries can be performed directly on the same datastore, it is also a very common pattern to use the queries
+to create *projections* of the write-side's events and store them into a separate datastore which is optimised for more
+complex queries. This architectural pattern of projecting the data into a query optimised datastore, with possibly some
+transformation or canculations along the way is the core use-case and recommended style of using Akka Persistence Query
+- pulling out of one Journal and storing into another one.
```

Comment:
Feels like this section is almost telling the same thing as the previous section.
I like the first one, see if something is unique here and if it can be incorporated in the first.